src/producer.rs

```diff
@@ -119,34 +120,74 @@ fn zip_producer(tmp_dir: &Path, zip_files: &[&String], queue: &WorkQueue, ignore
-                        if num != 0 {
-                            fs::hard_link(&physical_gcno_path, &gcno_path).expect(format!("Failed to create hardlink {}", gcno_path.display()).as_str());
-                        }
-
-                        let gcda_path = path.with_file_name(format!("{}_{}.gcda", stem, num + 1));
-
-                        extract_file(&mut gcda_file, &gcda_path);
-
-                        queue.push(Some(WorkItem {
-                            format: ItemFormat::GCNO,
-                            item: ItemType::Path(gcno_path),
-                            name: gcda_archive_name.to_string(),
-                        }));
+                        match gcno_buf_opt {
+                            Some(ref gcno_buf) => {
+                                let mut gcda_buf: Vec<u8> = Vec::new();
+                                gcda_file.read_to_end(&mut gcda_buf).expect("Failed to read gcda file");
+                                let gcno_stem = path.with_file_name(format!("{}_{}", stem, num + 1));
+                                let gcno_stem = gcno_stem.to_str().expect("Failed to create stem file string");
+
+                                queue.push(Some(WorkItem {
+                                    format: ItemFormat::GCNO,
+                                    item: ItemType::Buffers(GcnoBuffers {stem: gcno_stem.to_string(),
+                                                                         gcno_buf: Arc::clone(gcno_buf),
+                                                                         gcda_buf: gcda_buf}),
+                                    name: gcda_archive_name.to_string(),
+                                }));
+                            },
+                            None => {
+                                // Create symlinks.
+                                if num != 0 {
+                                    fs::hard_link(&physical_gcno_path, &gcno_path).expect(format!("Failed to create hardlink {}", gcno_path.display()).as_str());
+                                }
+
+                                let gcda_path = path.with_file_name(format!("{}_{}.gcda", stem, num + 1));
+
+                                extract_file(&mut gcda_file, &gcda_path);
+
+                                queue.push(Some(WorkItem {
+                                    format: ItemFormat::GCNO,
+                                    item: ItemType::Path(gcno_path),
+                                    name: gcda_archive_name.to_string(),
+                                }));
+                            }
+                        };
                     } else if num == 0 && !ignore_orphan_gcno {
-                        queue.push(Some(WorkItem {
-                            format: ItemFormat::GCNO,
-                            item: ItemType::Path(gcno_path),
-                            name: gcda_archive_name.to_string(),
-                        }));
+                        match gcno_buf_opt {
```

Comment:
It's a pity having to repeat this code twice, maybe we can refactor a bit to avoid it, but we can do it in the future.

Reply:
+1 let's do that later